lib/elixir/lib/dynamic_supervisor.ex

```diff
@@ -388,7 +388,7 @@ defmodule DynamicSupervisor do
           Supervisor.child_spec()
           | {module, term}
           | module
-          | (old_erlang_child_spec :: :supervisor.child_spec())
+          | Supervisor.old_erlang_child_spec()
```

Comment:
I would definitely prefer to refer to Erlang docs here instead instead of introducing such type.